client.go

```diff
@@ -78,8 +85,14 @@ func (c *client) serve(conn net.Conn) error {
 		default:
 			cmd, err = c.readCommand()
 			if err != nil {
-				zap.L().Error("read command failed", zap.String("addr", c.cliCtx.RemoteAddr),
-					zap.Int64("clientid", c.cliCtx.ID), zap.Error(err))
+				if err == io.EOF {
+					zap.L().Info("close connection", zap.String("addr", c.cliCtx.RemoteAddr),
+						zap.Int64("clientid", c.cliCtx.ID))
+					err = nil
```

Comment:
I suggest to just return here. Maybe you could put `c.conn.Close()` somewhere before this line

Reply:
OK